srt-protocol/src/protocol/sender/mod.rs

```diff
@@ -256,13 +254,13 @@ impl Sender {
                    self.congestion_control.window_size(),
                    self.transmit_buffer.next_sequence_number - self.congestion_control.window_size());
 
-            return WaitUntilAck;
+            return;
         } else if let Some(p) = self.pop_transmit_buffer() {
             self.send_data(p, now);
-        } else if self.close_requested {
-            // this covers the niche case of dropping the last packet(s)
-            if let Some(dp) = self.send_buffer.front().cloned() {
-                self.send_data(dp, now);
+        } else if self.status.should_drain() && self.send_buffer.len() == 1 {
+            if let Some(packet) = self.send_buffer.pop() {
+                self.send_data(packet, now);
+                self.lr_acked_packet = self.transmit_buffer.next_sequence_number;
```

Comment:
Again here

Reply:
I intended to at least remove this redundancy.